File: practical-assignment/src/main/java/com/project/practicalassignment/entity/Story.java
package com.project.practicalassignment.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Table(name = "Story")
public class Story {
    @Id
    @Column(name = "Story_Id")
    private long storyId;
    @Column(name = "Title")
    private String title;
    @Column(name = "Description")
    private String description;
    @OneToOne
    @JoinColumn(name = "Status_Id", referencedColumnName = "Status_Id")
    private Status status;
    @OneToOne
    @JoinColumn(name = "Priority_Id", referencedColumnName = "Priority_Id")
    private Priority priority;
    @OneToOne
    @JoinColumn(name = "Story_Type_Id", referencedColumnName = "Story_Type_Id")
    private StoryType type;
    @Column(name = "Created_On")
    private LocalDate createdDate;
}

File: Angular-Frontend/Angular-Server/src/app/model/story.ts
export class Story {

    // id: Number | undefined;
    title: string  = "";
    description: string  = "";
    status: {
        statusName: any;
        statusId: number;
    }  = {
        statusId: 0,
        statusName: ""
    }
    priority: {
        priorityLevel: any;
        priorityId: number
    } = {
        priorityId: 0,
        priorityLevel: ""
    };
    type: {
        storyType: any;
        storyTypeId: number
    } = {
        storyTypeId: 0,
        storyType: ""
    }
    createdDate: any | undefined;
    createdBy: {
        id: number;
        password: string;
        reportsTo: string;
        role: {
            id: number;
            role: string;
        };
        team: {
            id: number;
            name: string;
        };
        username: string;
    } = {
        id: 0,
        password: "",
        reportsTo: "",
        role: {
            id: 0,
            role: ""
        },
        team: {
            id: 0,
            name: ""
        },
        username: ""
    }
    assignedTo: {
        id: number;
        password: string;
        reportsTo: string;
        role: {
            id: number;
            role: string;
        };
        team: {
            id: number;
            name: string;
        };
        username: string;
    } = {
        id: 0,
        password: "",
        reportsTo: "",
        role: {
            id: 0,
            role: ""
        },
        team: {
            id: 0,
            name: ""
        },
        username: ""
    };
    team: {
        id: number;
        name: string;
    } = {
        id: 0,
        name: ""
    }
}
